lib/sendgrid/helpers/helpers.rb

```diff
@@ -0,0 +1,24 @@
+module SendGrid
+  module Helpers
+
+    # this method name is not accurate.  This method returns a hash of the non-nil instance variables
+    # TODO: update all uses to use :to_hash
+
+    def to_json
+      return nil if nil?
+
+      ret = {}
+      instance_variables.each do |attribute|
+        val = instance_variable_get(attribute)
+        val = val.strip if val.class == String
+        next if val.class == String && val.empty?
+        ret[attribute.to_s.tr('@', '')] = val unless val.nil?
+      end
```

Comment:
It might make for cleaner syntax to first convert all instance variables into an array, which can easily be converted into a hash with a `to_h` on `Array`. This should work well with ruby 2.2 and above that is supported by this gem. It's unfortunate in the code example below we have to add the `@` back into the `instance_variable_get` call, but `instance_variable_get` doesn't support leaving that off until Ruby 2.4.

Also, since a lot of these hashes are using reader methods that wrap around the instance variable, why don't you use `public_send` to get the value defined by the reader methods? This might get around some of your edge cases.

```rb
variable_names = instance_variables.map { |variable_name| variable_name.to_s.tr('@', '').to_sym }

hash = variable_names.map do |variable_name|
  [
    variable_name,
    respond_to?(variable_name) ? public_send(variable_name) : instance_variable_get(:"@#{variable_name}")
  ]
end.to_h

hash.delete_if { |_, value| value.to_s.empty? }
```